### PR TITLE
`aws_smithy_http_server::*` should be re-exported from generated crates

### PR DESCRIPTION
**Pull request by drganjoo**
Many customers have reported the need to explicitly depend on the `aws-smithy-http-server` crate in their service handler code. When they re-generate the crate, they often encounter version mismatches between the `aws-smithy-http-server` crate used in the generated code and the one they use in their service handler. This version discrepancy leads to compilation errors, requiring them to manually adjust the crate versions, which adds friction to their development workflow.

To resolve this issue, we now re-export all relevant types from `aws-smithy-http-server` within the generated crates. By doing so, customers can use these re-exported types directly, eliminating the need to depend on `aws-smithy-http-server` in their handler code.

Additionally, the generated crates no longer have the `aws-lambda` feature flag enabled by default. This prevents the `aws-lambda` feature from being automatically enabled in `aws-smithy-http-server` when the SDK is not intended for AWS Lambda.

### CODE REVIEW

**Review by david-perez — APPROVED** [review]
LGTM!

**Comment by david-perez** [comment]
We should announce this in the changelog.

File: codegen-server/src/main/kotlin/software/amazon/smithy/rust/codegen/server/smithy/customizations/ServerRequiredCustomizations.kt [hunk 1]
```diff
@@ -51,11 +51,19 @@ class ServerRequiredCustomizations : ServerCodegenDecorator {
         rustCrate.mergeFeature(
             Feature(
                 "aws-lambda",
-                true,
+                false,
```
Comment (david-perez) [inline codegen-server/src/main/kotlin/software/amazon/smithy/rust/codegen/server/smithy/customizations/ServerRequiredCustomizations.kt:54]: Why this change?

Comment (drganjoo) [reply codegen-server/src/main/kotlin/software/amazon/smithy/rust/codegen/server/smithy/customizations/ServerRequiredCustomizations.kt:54]: I changed the default feature flag to `false` so that if the generated SDK is not intended for AWS Lambda, it won't automatically enable the `aws-lambda` feature on the `aws-smithy-http-server` crate.
